### ISSUE TITLE
Crashlytics Gradle plugin is not compatible with configuration cache

### ISSUE THREAD

**Issue by maxsav** [firebase/firebase-android-sdk#6012]
### [READ] Step 1: Are you in the right place?

Issues filed here should be about bugs in __the code in this repository__. 
If you have a general question, need help debugging, or fall into some
other category use one of these other channels:

  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the firebase tag.
  * For general Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For help troubleshooting your application that does not fall under one
    of the above categories, reach out to the personalized
    [Firebase support channel](https://firebase.google.com/support/).

### [REQUIRED] Step 2: Describe your environment

  * Android Studio version: not related to the bug
  * Gradle version: 8.8
  * Firebase Component: Crashlytics Gradle plugin
  * Component version: 3.0.1

#### Steps to reproduce:

The gradle configuration cache is not supported by the task `com.google.firebase.crashlytics.buildtools.gradle.tasks.GenerateSymbolFileTask`

#### Gradle logs:
```
Task `:app:generateCrashlyticsSymbolFileRelease` of type `com.google.firebase.crashlytics.buildtools.gradle.tasks.GenerateSymbolFileTask`: invocation of 'Task.project' at execution time is unsupported.
See https://docs.gradle.org/8.8/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution
Invocation of 'Task.project' by task ':app:generateCrashlyticsSymbolFileRelease' at execution time is unsupported.
```

#### Related stacktrace:
```
org.gradle.api.InvalidUserCodeException: Invocation of 'Task.project' by task ':app:generateCrashlyticsSymbolFileRelease' at execution time is unsupported.
	at org.gradle.api.DefaultTask.getProject(DefaultTask.java:59)
        ...
	at com.google.firebase.crashlytics.buildtools.gradle.tasks.GenerateSymbolFileTask.resolveBreakpadBinary(GenerateSymbolFileTask.kt:152)
	at com.google.firebase.crashlytics.buildtools.gradle.tasks.GenerateSymbolFileTask.generateSymbolFiles(GenerateSymbolFileTask.kt:70)
        ...
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

**Comment by maxsav** [comment]
The reproducible error is here https://github.com/maxsav/firebase_crashlytics_configuration_cache_issue
You can see logs here https://github.com/maxsav/firebase_crashlytics_configuration_cache_issue/actions/runs/9378303532/job/25821294021

**Comment by lehcar09** [comment]
Thank you for the MCVE you shared @maxsav. I was able to reproduce the issue. It seems that the issue occurs only for `symbolGeneratorType = “breakpad”` and not for type `csym`. I’ll notify our engineers on this and see what we can do here. Thanks!

**Comment by lehcar09** [comment]
Hey @maxsav, according to our Eng, based on the investigation, the issue is caused by some code that was inadvertently moved into the task action to expedite the configuration process, which  accidentally broke the caching. 

The issue has been fixed, and it will be included in the next release in a couple weeks. We'll let you know when the fix is released. 

Thanks again!

**Comment by mrober** [comment]
This fix was released in version 3.0.2.
